draft-ietf-quic-transport.md

```diff
@@ -338,9 +340,14 @@ Packet Number:
 : The Packet Number is a 32-bit field that follows the two connection IDs.
   {{packet-numbers}} describes the use of packet numbers.
 
+Payload Length:
+
+: The length of the Payload field in octets, encoded as a variable-length
+  integer ({{integer-encoding}}).
+
```

Comment:
I came to think a bit about the variable payload length:

It makes it difficult to build a packet when you don't know the length in advance. Of course you can always just start with enough space. The intial packet starts with 1200 bytes (not sure about 0-RTT), and I don't think it can be more than 64K? So wouldn't a fixed 2 byte length field make more sense here?

Reply:
The variable length is primarily for consistency.  There are a few cases where packets will be small enough to benefit from the 6-bit length (Finished primarily), but the main benefit is that this is the same code throughout.  Better if we can manage to make the packet number encoding variable length as well (but that interacts in interesting ways with #1079, so that's not as clear a win as this).